bundles/org.openhab.core/src/main/java/org/openhab/core/items/ManagedItemProvider.java

```diff
@@ -118,6 +128,16 @@ public ManagedItemProvider(final @Reference StorageService storageService,
         return memberNames;
     }
 
+    private Set<Item> getMembers(GroupItem groupItem, Collection<Item> allItems) {
+        Set<Item> members = new HashSet<>();
+        for (Item item : allItems) {
```

Comment:
Wouldn't it be easier to iterate over `group.getMembers()`?

Reply:
If that would be possible, the other recursive function in the `ManagedItemProvider `could use that as well. But `group.getMembers()` is empty since you're fetching the Item at the start of `Remove(string key)` from the `StorageService`.

Reply:
I think digging into it and research why group members are empty needs too much time. Considering how often a group will be deleted it should be okay for now to accept the iteration over all items. Wdyt?

Reply:
No need to research that, I think. There's an `initializeItem(Item item)` in the `ItemRegistry` where members get filled in. If I want to have an item filled in with members, I would need to get that item from the `ItemRegistry` on removal or the remove function would need to be adjusted to contain the item instead of the key. 

I can't use the `ItemRegistry` removal event since it gets called too whenever a provider get disposed (https://github.com/openhab/openhab-core/blob/master/bundles/org.openhab.core/src/main/java/org/openhab/core/common/registry/AbstractRegistry.java#L640) . Making a dedicated service with a reference to the `ManagedItemProvider` and `ItemRegistry`, so I can listen on `ManagedItemProvider` removals and then fetch the item from `ItemRegistry` would only work if my service is being called before the listener of `ItemRegistry`, which cannot be guaranteed.

Reply:
Thank you for explanation. Let's go for the current solution.